File: ferda/src/Modules/Core/MiningProcessor/Service.cs
using Ferda.Guha.MiningProcessor.Miners;
using Ice;

namespace Ferda.Guha.MiningProcessor
{
    public class Service : LocalObjectImpl, IceBox.Service
    {
        private ObjectAdapter _adapter;

        #region ServiceOperationsNC_ Members

        public void start(string name, Communicator communicator, string[] args)
        {
            _adapter = communicator.createObjectAdapter(name);
            _adapter.add(new MiningProcessorFunctionsI(),
                         Util.stringToIdentity("Ferda.Guha.MiningProcessor.MiningProcessorFunctions"));
            ObjectFactory factory = new ObjectFactory();
            ObjectFactory.addFactoryToCommunicator(communicator, factory);
            _adapter.activate();
        }

        public void stop()
        {
            _adapter.deactivate();
        }

        #endregion
    }
}

File: ferda/src/Modules/FerdaOWLParser/Ferda/OntologyRelated/Service.java
package Ferda.OntologyRelated;

// class for setting up the ICE communication between Ferda box ontology and FerdaOWLParser
public class Service extends Ice.LocalObjectImpl implements IceBox.Service
{
	private Ice.ObjectAdapter _adapter;
	
	//function needed by ICE
	public void start(String name, Ice.Communicator communicator, String[] args)
	{
	    _adapter = communicator.createObjectAdapter(name + "Adapter");
	    _adapter.add(new OWLParserI(), Ice.Util.stringToIdentity("Ferda.OntologyRelated.OWLParser"));
	    
	    _adapter.activate();
	}

	//function needed by ICE
	public void stop()
	{
		_adapter.deactivate();
	}

}
